src/mods/core.js

```diff
@@ -112,33 +165,165 @@ idrinth.core = {
             body: content
         } );
     },
+    /**
+     *
+     * @type {object}
+     */
+    timeouts: {
+        /**
+         *
+         * @type {object}
+         */
+        next: null,
+        /**
+         *
+         * @type {object}
+         */
+        list: { },
+        /**
+         * makes sure the timeout is removed when called next
+         * @param {string} identifier
+         * @returns {undefined}
+         */
+        remove: function ( identifier ) {
+            'use strict';
+            if ( idrinth.core.timeouts.list[identifier] !== undefined ) {
+                idrinth.core.timeouts.list[identifier].interval = false;
+                idrinth.core.timeouts.list[identifier].func = function () {
+                };
+            }
+        },
+        /**
+         *
+         * @param {string} identifier
+         * @param {function} func
+         * @param {int} time in milliseconds
+         * @param {Boolean} [false] isInterval
+         * @returns {undefined}
+         */
+        add: function ( identifier, func, time, isInterval ) {
+            'use strict';
+            var date = new Date ();
+            idrinth.core.timeouts.list[identifier] = {
+                func: func,
+                next: date.getTime () + date.getMilliseconds () / 1000 + time / 1000,
+                duration: time,
+                interval: !!isInterval
+            };
+        },
+        /**
+         * activates all relevant timeouts and intervals
+         * @returns {undefined}
+         */
+        process: function () {
+            'use strict';
+            var date = ( new Date () ).getTime () + ( new Date () ).getMilliseconds () / 1000;
+            var min = 10;
+            /**
+             *
+             * @param {Number} durationLeft
+             * @param {Number} minDuration
+             * @returns {Number}
+             */
+            var getVal = function ( durationLeft, minDuration ) {
+                if ( durationLeft < 0.1 ) {
+                    return 0.1;
+                }
+                return durationLeft < minDuration ? durationLeft : minDuration;
+            };
+            for (var property in idrinth.core.timeouts.list) {
+                if ( idrinth.core.timeouts.list.hasOwnProperty ( property ) ) {
+                    if ( date >= idrinth.core.timeouts.list[property].next ) {
+                        try {
+                            idrinth.core.timeouts.list[property].func ();
+                            if ( idrinth.core.timeouts.list[property].interval ) {
+                                min = getVal ( idrinth.core.timeouts.list[property].duration, min );
+                                idrinth.core.timeouts.list[property].next = date + idrinth.core.timeouts.list[property].duration / 1000;
+                            } else {
+                                delete idrinth.core.timeouts.list[property];
```

Comment:
You might not need to delete property? You could set value to undefined and verify if it's not undefined? :)

>  delete idrinth.core.timeouts.list[property];

Won't your timeout still be going, even if you removed it from the list?

Just a suggestion, I see you have interval option to repeat interval. Could be good to have property, which will set number of times interval would be repeated and the delete itself.